computation_graph/visualization.py

```diff
@@ -0,0 +1,67 @@
+from typing import Text
+
+import pygraphviz as pgv
+
+from computation_graph import base_types, graph
+
+
+def _get_edge_exceptions_string(edge: base_types.ComputationEdge) -> Text:
+    return f" [{','.join(map(repr,edge.allowed_exceptions))}]"
+
+
+def _get_edge_label(edge: base_types.ComputationEdge):
+    if edge.key in (None, "args"):
+        return "" + _get_edge_exceptions_string(edge)
+
+    if edge.key == "first_input":
+        return f"{edge.priority}" + _get_edge_exceptions_string(edge)
+
+    return (edge.key or "") + _get_edge_exceptions_string(edge)
+
+
+def _get_node_shape(node: base_types.ComputationNode):
+    if node.name == "first":
+        return "triangle"
+
+    return "ellipse"
+
+
+def _add_single_node(
+    pgv_graph: pgv.AGraph,
+    edges: base_types.GraphType,
+    node: base_types.ComputationNode,
+):
+    pgv_graph.add_node(
+        graph.infer_node_id(edges, node), label=node, shape=_get_node_shape(node)
```

Comment:
`graph.infer_node_id(edges, node)` repeats 3 times, use an assignment to deduplicate